x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDataFrameAnalyticsInfoAction.java

```diff
@@ -0,0 +1,145 @@
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.core.ml.action.DataFrameAnalyticsInfoAction;
+import org.elasticsearch.xpack.core.ml.action.PutDataFrameAnalyticsAction;
+import org.elasticsearch.xpack.core.ml.dataframe.info.FieldSelection;
+import org.elasticsearch.xpack.core.ml.dataframe.info.MemoryEstimation;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+import org.elasticsearch.xpack.ml.MachineLearning;
+import org.elasticsearch.xpack.ml.dataframe.extractor.DataFrameDataExtractorFactory;
+import org.elasticsearch.xpack.ml.dataframe.extractor.ExtractedFieldsDetector;
+import org.elasticsearch.xpack.ml.dataframe.extractor.ExtractedFieldsDetectorFactory;
+import org.elasticsearch.xpack.ml.dataframe.process.MemoryUsageEstimationProcessManager;
+import org.elasticsearch.xpack.ml.extractor.ExtractedFields;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Retrieves info for the given data frame analytics spec.
+ * Redirects to a different node if the current node is *not* an ML node.
+ */
+public class TransportDataFrameAnalyticsInfoAction
+    extends HandledTransportAction<PutDataFrameAnalyticsAction.Request, DataFrameAnalyticsInfoAction.Response> {
+
+    private final TransportService transportService;
+    private final ClusterService clusterService;
+    private final NodeClient client;
+    private final MemoryUsageEstimationProcessManager processManager;
+
+    @Inject
+    public TransportDataFrameAnalyticsInfoAction(TransportService transportService,
+                                                 ActionFilters actionFilters,
+                                                 ClusterService clusterService,
+                                                 NodeClient client,
+                                                 MemoryUsageEstimationProcessManager processManager) {
+        super(DataFrameAnalyticsInfoAction.NAME, transportService, actionFilters, PutDataFrameAnalyticsAction.Request::new);
+        this.transportService = transportService;
+        this.clusterService = Objects.requireNonNull(clusterService);
+        this.client = Objects.requireNonNull(client);
+        this.processManager = Objects.requireNonNull(processManager);
+    }
+
+    @Override
+    protected void doExecute(Task task,
+                             PutDataFrameAnalyticsAction.Request request,
+                             ActionListener<DataFrameAnalyticsInfoAction.Response> listener) {
+        DiscoveryNode localNode = clusterService.localNode();
```

Comment:
I wonder if this should be license restricted?

Reply:
We might help people by checking the license here, because explain is a precursor to create/start, and it would avoid frustration if we tell people they're not licensed as early as possible in the create/start sequence

Reply:
Well spotted @benwtrent ! I'll add a license check.